pkg/proxy/types.go

```diff
@@ -42,3 +43,33 @@ type ServicePortName struct {
 func (spn ServicePortName) String() string {
 	return fmt.Sprintf("%s:%s", spn.NamespacedName.String(), spn.Port)
 }
+
+// ServicePort is an interface which abstracts a service information.
```

Comment:
After this change serviceInfo will not exist anymore right? I'm not sure what the best way to change this comment is but it definitely should not include reference to serviceInfo